build.sh

```diff
@@ -44,31 +38,7 @@ if [ "$BUILD_CCF_FROM_SOURCE" = "ON" ]; then
     popd
 fi
 
-if [ "$ENABLE_CLANG_TIDY" = "ON" ]; then
-    if ! command -v clang-tidy &> /dev/null && ! command -v clang-tidy-18 &> /dev/null; then
-        echo "clang-tidy could not be found, please install it, e.g. tdnf install -y clang-tools-extra-devel"
-        exit 1
-    fi
-fi
-
-root_dir=$(pwd)
-install_dir=/tmp/scitt
-
-mkdir -p $install_dir
-
-# Note: this is a development build.
-# See docker/ for a non-development build.
-CC="$CC" CXX="$CXX" \
-    cmake -GNinja -B build/app \
-    -DCMAKE_BUILD_TYPE="${CMAKE_BUILD_TYPE}" \
-    -DCOMPILE_TARGET="${PLATFORM}" \
-    -DBUILD_TESTS="${BUILD_TESTS}" \
-    -DCMAKE_INSTALL_PREFIX=$install_dir \
-    -DENABLE_CLANG_TIDY="${ENABLE_CLANG_TIDY}" \
-    "$root_dir/app"
-
-ninja -C build/app ${NINJA_FLAGS} --verbose
-ninja -C build/app ${NINJA_FLAGS} install
-
-echo "List of installed files in SCITT_DIR $install_dir"
-ls -R $install_dir
+cd app
+cmake --workflow --preset "$PLATFORM" --fresh
```

Comment:
Ok I see, this changes the default build from Release to Debug, which is fine, but I guess it's important to check the release pipelines are all setting this explicitly and not getting debug builds. Also the BUILD_CCF_FROM_SOURCE is doing Debug/ASAN, which is fine, but can maybe be renamed to BUILD_DEBUG_CCF_FROM_SOURCE to avoid accidents. Quite separate from the main goal of the PR, but since this change is happening here, I thought I'd mention it anyway :)

Reply:
when releasing we build in docker containers without invoking this script, otherwise the var renamed